app/src/androidTest/java/ch/epfl/sdp/appart/ResetPasswordTest.java

```diff
@@ -0,0 +1,90 @@
+package ch.epfl.sdp.appart;
+
+import androidx.test.espresso.intent.Intents;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.concurrent.ExecutionException;
+
+import javax.inject.Inject;
+
+import ch.epfl.sdp.appart.login.LoginService;
+import ch.epfl.sdp.appart.user.LoginActivity;
+import ch.epfl.sdp.appart.user.ResetActivity;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.junit.Assert.assertNull;
+
+@HiltAndroidTest
+public class ResetPasswordTest {
+
+    @Rule(order = 0)
+    public HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule(order = 1)
+    public ActivityScenarioRule<ResetActivity> resetPasswordActivityRule = new ActivityScenarioRule<>(ResetActivity.class);
+
+    @Inject
+    LoginService loginService;
+
+    @Before
+    public void init() {
+        hiltRule.inject();
+        Intents.init();
+        loginService.useEmulator("10.0.2.2", 9099);

```

Comment:
What about defining a LoginServiceEmulator and a MockLoginService ? I personally think its better because once we inject the dependency, I think the object should not change.

Reply:
I like this solution of having the two.